protocols/stream/src/control.rs

```diff
@@ -103,6 +103,17 @@ impl fmt::Display for OpenStreamError {
     }
 }
 
+impl std::error::Error for OpenStreamError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use OpenStreamError::*;
+
+        match self {
+            Io(error) => Some(error),
+            _ => None,
+        }
+    }
```

Comment:
nit:
```suggestion
        match self {
            Self::Io(error) => Some(error),
            _ => None,
        }
    }
```
specially since it's just one variant